presto-main/src/main/java/com/facebook/presto/execution/MemoryRevokingScheduler.java

```diff
@@ -244,10 +251,12 @@ public Long mergeResults(List<Long> childrenResults)
     private void requestRevoking(MemoryPool memoryPool, Collection<SqlTask> sqlTasks, long remainingBytesToRevoke)
     {
         AtomicLong remainingBytesToRevokeAtomic = new AtomicLong(remainingBytesToRevoke);
+        Ordering<SqlTask> sqlTaskOrdering = spillingStrategy == TaskSpillingStrategy.ORDER_BY_CREATE_TIME ? ORDER_BY_CREATE_TIME : ORDER_BY_REVOCABLE_MEMORY;
```

Comment:
nit: consider use a switch? So you can throw `UnsupportedOperationException` for enum values not applicable.

Reply:
I also added checkArgument in constructor to make sure this wont happen.